SharedTools/Download-GitExtensions.ps1

```diff
@@ -0,0 +1,115 @@
+param([string] $Version, [string] $Source = "github")
+
+if (-not($Version)) 
+{ 
+    Throw "Parameter -Version is required";
+}
+
+if (-not($Source -eq "github") -and -not($Source -eq "appveyor"))
+{
+	Throw "Parameter -Source must be either 'github' or 'appveyor'";
+}
+
+Set-Location $PSScriptRoot
+
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$ExtractRootPath = '../references'
+$AssetToDownloadName = 'GitExtensions-' + $Version + '.zip';
+$AssetToDownloadUrl = $null;
+
+$DownloadName = [System.IO.Path]::GetFileName($AssetToDownloadName);
+$DownloadFilePath = [System.IO.Path]::Combine($ExtractRootPath, $DownloadName);
+if (Test-Path $DownloadFilePath)
+{
+	Write-Host ('Download "' + $DownloadName + '" already exists.');
+	Pop-Location
+	exit;
```

Comment:
It is a good idea to have different return values, so if this script is used by other scripts, they can reason about the exit state

```suggestion
	exit -1;
```